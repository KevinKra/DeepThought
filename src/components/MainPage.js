import React, { Component } from "react";
import CardSlideshow from "./CardSlideshow";
import NavBar from "./NavBar";
import UserSection from "./UserSection";
import "../styles/_MainPage.scss";

class MainPage extends Component {
  render() {
    return (
      <main className="mainPage">
        <NavBar history={this.props.history} />
        <CardSlideshow />
        <UserSection />
      </main>
    );
  }
}

export default MainPage;
