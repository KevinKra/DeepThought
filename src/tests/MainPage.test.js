import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import MainPage from "../components/MainPage";

const defaultState = {
  displayReviewPage: false,
  displayNotFoundPage: false,
  displayMainPage: true
};

const displayNotFound = {
  displayReviewPage: false,
  displayNotFoundPage: true,
  displayMainPage: false
};

const displayReviewPage = {
  displayReviewPage: true,
  displayNotFoundPage: false,
  displayMainPage: false
};

const displayMainPage = {
  displayReviewPage: false,
  displayNotFoundPage: false,
  displayMainPage: true
};

describe("<MainPage />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MainPage />);
  });
  it("should match the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should match default state", () => {
    expect(wrapper.state()).toEqual(defaultState);
  });
  it("renderNotFoundPage should update local state", () => {
    expect(wrapper.state()).toEqual(defaultState);
    wrapper.instance().renderNotFoundPage();
    expect(wrapper.state()).toEqual(displayNotFound);
  });
  it("renderReviewPage should update local state", () => {
    expect(wrapper.state()).toEqual(defaultState);
    wrapper.instance().renderReviewPage();
    expect(wrapper.state()).toEqual(displayReviewPage);
  });
  it("renderMainPage should update local state", () => {
    expect(wrapper.state()).toEqual(defaultState);
    wrapper.instance().renderMainPage();
    expect(wrapper.state()).toEqual(displayMainPage);
  });
  // it("if local state displayReviewPage is true, render <ReviewPage />", () => {
  //   wrapper.instance().renderReviewPage();
  //   expect(wrapper.state()).toEqual(displayReviewPage);
  //   how to test a function expression inside of render() method
  //   expect(wrapper.instance().renderPath());
  // });
  // it("if local state displayNotFoundPage is true, render <NotFound />", () => {});
  // it("if local state displayMainPage is true, render mainPage", () => {});
});
