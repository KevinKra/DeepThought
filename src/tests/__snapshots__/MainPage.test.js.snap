// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> should match the snapshot 1`] = `
<Fragment>
  <div
    className="mainPage"
  >
    <NavBar />
    <CardSlideshow
      renderNotFound={[Function]}
      renderReview={[Function]}
    />
    <UserSection />
  </div>
</Fragment>
`;
